python/tvm/relay/ir_pass.py

```diff
@@ -169,3 +169,18 @@ def graph_equal(lhs, rhs):
       True iff lhs is data-flow equivalent to rhs.
     """
     return bool(_make._graph_equal(lhs, rhs))
+
+def expr_hash(expr):
```

Comment:
we need a good name for this, expr_hash is a bit generic. Maybe structural_hash/? Most users get used to same function name being overloaded for types, so we could use it for both type and Expr